tasks/gitlab_helpers.py

```diff
@@ -55,31 +57,43 @@ def create_gitlab_annotations_report(ci_job_id: str, ci_job_name: str):
             {
                 "external_link": {
                     "label": "CI Visibility: This job instance",
-                    "url": get_link_to_job_id(ci_job_id),
+                    "url": get_link_to_pipeline_job_id(ci_job_id),
+                }
+            },
+            {
+                "external_link": {
+                    "label": "CI Visibility: This job test runs",
+                    "url": get_link_to_test_runs(ci_job_id),
                 }
             },
             {
                 "external_link": {
                     "label": "CI Visibility: This job on main",
-                    "url": get_link_to_job_on_main(ci_job_name),
+                    "url": get_link_to_pipeline_job_on_main(ci_job_name),
+                }
+            },
+            {
+                "external_link": {
+                    "label": "CI Visibility: This job test runs on main",
+                    "url": get_link_to_test_runs_on_main(ci_job_name),
                 }
             },
         ]
     }
 
 
-def get_link_to_job_id(job_id: str):
+def get_link_to_pipeline_job_id(job_id: str):
     query_params = {
         "ci_level": "job",
         "@ci.job.id": job_id,
         "@ci.pipeline.name": "DataDog/datadog-agent",
     }
     query_string = to_query_string(query_params)
```

Comment:
why don't we use `urllib.parse.urlencode` here? It's normally [already quoted](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode:~:text=By%20default%2C%20quote_plus()%20is%20used%20to%20quote%20the%20values)

Reply:
I think because the entire string is a query value, so characters like `&` need to be encoded too.

Reply:
From `urllib.parse.urlencode`

>An alternate function that can be passed as quote_via is [quote()](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote), which will encode spaces as %20 and not encode ‘/’ characters. For maximum control of what is quoted, use quote and specify a value for safe.

I could use `quote(urlencode(query_params))`, didn't know it!

Reply:
Spent some time on this, we cannot use `urlencode` as the query format in DD requires `key:value` format, while `urlencode` formats to `key=value`

See an example of raw query: 

```
query=ci_level%3Ajob%20%40ci.pipeline.name%3ADataDog%2Fdatadog-agent&agg_m=count&agg_m_source=base&agg_t=count&fromUser=false&index=cipipeline&start=1718795544509&end=1719400344509&paused=false
```

The `query` parameter is one of the global query params. Internally it contains

```
ci_level%3Ajob%20%40ci.pipeline.name%3ADataDog%2Fdatadog-agent
```

Converting back encoded chars it becomes

```
ci_level:job @ci.pipeline.name:DataDog/datadog-agent
```

I could not find a way to change the default separators in `urlencode`, thus we need the custom function